File: Data Structures & Algorithms/binary-tree-diameter/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int res=0;
    public int diameterOfBinaryTree(TreeNode root) {
        maxdepth(root);
        return res;
    }
    public int maxdepth(TreeNode root){
        if(root==null) return 0;
        else{
            int leftmax=maxdepth(root.left);
            int rightmax=maxdepth(root.right);
            res=Math.max(res,leftmax+rightmax);
            return Math.max(leftmax,rightmax)+1;
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root==None:
            return []
        res=[]
        s=collections.deque([root])
        while s:
            size=len(s)
            l1=[]
            for i in range(size):
                curr=s.popleft()
                l1.append(curr.val)
                if curr.left!=None:
                    s.append(curr.left)
                if curr.right!=None:
                    s.append(curr.right)
            res.append(l1)
        return res


